src/runner/system_menu_helper.cpp

```diff
@@ -0,0 +1,155 @@
+#include "pch.h"
+#include "system_menu_helper.h"
+
+#include <interface/powertoy_module_interface.h>
+
+namespace {
+  constexpr int KSeparatorPos = 1;
+  constexpr int KNewItemPos = 2;
+
+  int GenerateItemId() {
+    static int generator = 0xDEADBEEF;
+    return ++generator;
+  }
+}
+
+SystemMenuHelper& SystemMenuHelperInstace()
+{
+  static SystemMenuHelper instance;
+  return instance;
+}
+
+void SystemMenuHelper::SetConfiguration(PowertoyModuleIface* module, const wchar_t* config)
+{
+  Reset(module);
+  std::vector<ItemInfo> info;
+  ParseConfiguration(std::wstring(config), info);
+  Configurations[module] = info;
+  for (auto& [window, modules] : ProcessedWindows) {
+    // Unregister module. After system menu is openned again, new configuration will be applied.
+    modules.erase(std::remove(std::begin(modules), std::end(modules), module), std::end(modules));
+  }
+}
+
+void SystemMenuHelper::RegisterAction(PowertoyModuleIface* module, HWND window, const wchar_t* name)
```

Comment:
This method is used to check/uncheck the menu item (and in the future it may be used to also do other things like adding a status icon in the menu title-bar for example), but it doesn't `register` an action, it processes the user action for selecting a menu item, so a better name would be `ProcessSelectedItem`.
Also `const wchar_t* name`  -> `const wchar_t* itemName`.